macros/audit.sql

```diff
@@ -55,21 +62,31 @@
 
 
 {% macro create_audit_schema() %}
-    create schema if not exists {{ logging.get_audit_schema() }}
+    {% do create_schema(api.Relation.create(
+        database=target.database,
+        schema=logging.get_audit_schema())
+    ) %}
 {% endmacro %}
 
 
 {% macro create_audit_log_table() -%}
 
+    {{ return(adapter.dispatch('create_audit_log_table', packages=['logging'])()) }}
+
+{% endmacro %}
+
+
+{% macro default__create_audit_log_table() -%}
+
     {% set required_columns = [
-       ["event_name", "varchar(512)"],
+       ["event_name", {{ dbt_utils.type_string() }}],
        ["event_timestamp", dbt_utils.type_timestamp()],
-       ["event_schema", "varchar(512)"],
-       ["event_model", "varchar(512)"],
-       ["event_user", "varchar(512)"],
-       ["event_target", "varchar(512)"],
+       ["event_schema", {{ dbt_utils.type_string() }}],
+       ["event_model", {{ dbt_utils.type_string() }}],
+       ["event_user", {{ dbt_utils.type_string() }}],
+       ["event_target", {{ dbt_utils.type_string() }}],
        ["event_is_full_refresh", "boolean"],
-       ["invocation_id", "varchar(512)"],
+       ["invocation_id", {{ dbt_utils.type_string() }}],
```

Comment:
```suggestion
       ["event_name", dbt_utils.type_string()],
       ["event_timestamp", dbt_utils.type_timestamp()],
       ["event_schema", dbt_utils.type_string()],
       ["event_model", dbt_utils.type_string()],
       ["event_user", dbt_utils.type_string()],
       ["event_target", dbt_utils.type_string()],
       ["event_is_full_refresh", "boolean"],
       ["invocation_id", dbt_utils.type_string()],
```
sorry if I led you astray with my previously suggested code here

Reply:
Good catch, updated the PR